tools/topology/sof-apl-src-pcm512x.m4

```diff
@@ -0,0 +1,71 @@
+#
+# Topology for generic Apollolake UP^2 with pcm512x codec and HDMI.
+#
+
+# Include topology builder
+include(`utils.m4')
+include(`dai.m4')
+include(`pipeline.m4')
+include(`ssp.m4')
+include(`hda.m4')
+
+# Include TLV library
+include(`common/tlv.m4')
+
+# Include Token library
+include(`sof/tokens.m4')
+
+# Include Apollolake DSP configuration
+include(`platform/intel/bxt.m4')
+
+DEBUG_START
+
+#
+# Define the pipelines
+#
+# PCM0 ----> src -----> SSP5 (pcm512x)
+#
+
+dnl PIPELINE_PCM_ADD(pipeline,
+dnl     pipe id, pcm, max channels, format,
+dnl     frames, deadline, priority, core)
+
+# Low Latency playback pipeline 1 on PCM 0 using max 2 channels of s32le.
+# Schedule 48 frames per 1000us deadline on core 0 with priority 0
+PIPELINE_PCM_ADD(sof/pipe-src-playback.m4,
+	1, 0, 2, s32le,
```

Comment:
@singalsu is the format correct here? Should it be s24le?

Reply:
Good catch, yes it needs to be fixed while we can't support it. I will need to do later for SRC similar 16/24/32 support as EQs have and place volume to pipeline to convert to DAI. But for now it's best to limit to only what works!

Reply:
done, though I had to edit also pipe-src-playback.m4 to get aplay to reject other than s24.